File: Answer/2154.将找到的值乘以-2.js
/*
 * @lc app=leetcode.cn id=2154 lang=javascript
 *
 * [2154] 将找到的值乘以 2
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function (nums, original) {
    var set = new Set();
    nums.forEach(num => {
        set.add(num);
    });
    while (set.has(original)) {
        original *= 2;
    }
    return original;
};
// @lc code=end

File: Answer/2154.将找到的值乘以-2.go
/*
 * @lc app=leetcode.cn id=2154 lang=golang
 *
 * [2154] 将找到的值乘以 2
 */

// @lc code=start
func findFinalValue(nums []int, original int) int {
    m := make(map[int]bool)
	for _, k := range nums {
		m[k] = true;
	}
	for ;; {
		if m[original] {
			original *= 2
		} else {
			break
		}
	}
	return original
}
// @lc code=end

